src/main/java/pl/javastart/task/Product.java

```diff
@@ -0,0 +1,20 @@
+package pl.javastart.task;
+
+public class Product {
+    Category category;
+    SpecialOffer specialOffer;
```

Comment:
to pole jest niepotrzebne, w ten sposób ograniczamy się do tego, że produkt może być tylko w ramach jednej promocji. W `SpecialOffer` masz już pole `Porduct` i to jest poprawne podejście, bo możemy sobie tworzyć oferty specjalne bez ograniczeń